### PR TITLE
frontend/enhance: calculate ride distance based on accuracy of the GPS

### PR DESCRIPTION
**Pull request by witcher-shailesh**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/LocationUpdateService.java [hunk 1]
```diff
@@ -155,6 +166,13 @@ public void onCreate() {
     public int onStartCommand(Intent intent, int flags, int startId) {
         /* Start the service if the driver is active*/
         startForeground(notificationServiceId, createNotification());
+        String startDistanceCalStr = intent.hasExtra("TRIP_STATUS") ? intent.getStringExtra("TRIP_STATUS") : null;
+        if (!isDistanceCalulation && startDistanceCalStr != null && startDistanceCalStr.equals("started")) {
```
Comment (Vignesh-772) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/LocationUpdateService.java:170]: Once we start the `fusedLocClientForDistanceCal`. we will not stop in again after ride ended? Because we are not making `isDistanceCalulation` to false. and we are removing removing location updates in onDestroy. so even after the ride ended we are reseting these two finalDistanceWithAcc, finalDistance variables. but we didn't stop the location updates so the distance will be calculated and accumulated with the next ride's distance.

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/LocationUpdateService.java [hunk 2]
```diff
@@ -287,57 +356,22 @@ private void updateDriverStatus(Boolean status) {
...
-                    respReader = new InputStreamReader(connection.getErrorStream());
-                    Log.d(LOG_TAG, "in error " + respReader);
-                } else {
-                    respReader = new InputStreamReader(connection.getInputStream());
-                    Log.d(LOG_TAG, "in 200 " + respReader);
+                MobilityCallAPI mobilityApiHandler = new MobilityCallAPI();
+                Map<String, String> baseHeaders = mobilityApiHandler.getBaseHeaders(context);
+                MobilityAPIResponse apiResponse = mobilityApiHandler.callAPI(orderUrl, baseHeaders);
+                if (!status) {
+                    updateStorage("LOCATION_STATUS", "PAUSE"); // TO-ASK:: Can we remove it, don't think it's being used
```
Comment (Vignesh-772) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/LocationUpdateService.java:367]: > TO-ASK:: Can we remove it, don't think it's being

this we can remove

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/LocationUpdateService.java [hunk 3]
```diff
@@ -862,6 +832,10 @@ private boolean isLocationEnabled() {
         return locationManager != null && LocationManagerCompat.isLocationEnabled(locationManager);
     }
 
+    private void calculateRideDistance(Location location){
+
```
Comment (Vignesh-772) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/LocationUpdateService.java:836]: is it added for any other use case. If not we can remove.

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/MyFirebaseMessagingService.java [hunk 4]
```diff
@@ -249,6 +249,12 @@ public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
                             startMainActivity();
                             break;
 
+                        case NotificationTypes.TRIP_STARTED:
+                            if (payload.get("show_notification").equals("true")) {
```
Comment (Vignesh-772) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/MyFirebaseMessagingService.java:253]: we can remove this already handled in default

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/TLSSocketFactory.java [hunk 5]
```diff
@@ -25,7 +25,7 @@
 import javax.net.ssl.TrustManager;
 import javax.net.ssl.X509TrustManager;
 
-class TLSSocketFactory extends SSLSocketFactory {
+public class TLSSocketFactory extends SSLSocketFactory {
```
Comment (Vignesh-772) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/TLSSocketFactory.java:28]: is this change required? It is not used.

File: Frontend/ui-driver/src/Helpers/Storage.purs [hunk 6]
```diff
@@ -98,6 +98,10 @@ data KeyStore = USER_NAME
                 | DRIVER_SUBSCRIBED
                 | KIOSK_LOCATIONS
                 | ENABLE_BLOCKING
+                | TRIP_DISTANCE_ACC
+                | TRIP_DISTANCE
+                | TRIP_STATUS
+                | TRIP_STARTED
```
Comment (Vignesh-772) [inline Frontend/ui-driver/src/Helpers/Storage.purs:104]: This is not used

**Review by Vignesh-772 — APPROVED** [review]

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]
Requested Changes

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/LocationUpdateService.java [hunk 7]
```diff
@@ -167,7 +185,52 @@ public int onStartCommand(Intent intent, int flags, int startId) {
...
+                                String accHolder = getValueFromStorage("ACCURACY_THRESHOLD");
+                                String accuracyThreshold = accHolder == null ? "100.0" : accHolder;
+                                if (prevLocation != null && prevAccLocation != null) {
+                                    distanceBtLatLng = (int) location.distanceTo(prevLocation);
+                                    finalDistance+=distanceBtLatLng;
+                                    if (checkLocationAcc(location, accuracyThreshold)) {
+                                        distanceBtLatLngWithAcc = (int) location.distanceTo(prevAccLocation);
+                                        finalDistanceWithAcc+=distanceBtLatLngWithAcc;
+                                        updateStorage("TRIP_DISTANCE_ACC", Integer.toString(finalDistanceWithAcc));
+                                        prevAccLocation = location;
```
Comment (max-keviv) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/LocationUpdateService.java:218]: do we need this `prevAccLocation = location;` as it is already defined in line 225

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/LocationUpdateService.java [hunk 8]
```diff
@@ -167,7 +185,52 @@ public int onStartCommand(Intent intent, int flags, int startId) {
...
+                fusedLocClientForDistanceCal.getCurrentLocation(Priority.PRIORITY_HIGH_ACCURACY, cancellationTokenSource.getToken())
+                        .addOnSuccessListener(location -> {
+                            if(location!=null) {
+                                isDistanceCalulation = true;
+                                int distanceBtLatLng = 0, distanceBtLatLngWithAcc = 0;
+                                String accHolder = getValueFromStorage("ACCURACY_THRESHOLD");
+                                String accuracyThreshold = accHolder == null ? "100.0" : accHolder;
+                                if (prevLocation != null && prevAccLocation != null) {
+                                    distanceBtLatLng = (int) location.distanceTo(prevLocation);
+                                    finalDistance+=distanceBtLatLng;
```
Comment (max-keviv) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/LocationUpdateService.java:213]: is finalDistance dependent on prevAccLocation value as if it is null it won't be upated
